src/gps/dateTime.c

```diff
@@ -133,3 +133,118 @@ tiny_millis_t getUptime() {
 int getUptimeAsInt() {
    return (int) getUptime();
 }
+
+void getDateTimeFromEpochMillis(DateTime *dateTime, millis_t millis) {
```

Comment:
On a limited stack environment (freeRTOS thread) we have to be careful what we allocated. By passing it by value, we first allocate it once, then it's allocated again within the function; then it's copied back to the caller.   So we have 2x stack usage as well as copying the values back to the caller.

I don't know if GCC will optimize this out; I'm going to assume it doesn't  unless there's definitive proof.